File: LeakyBucket.java
import java.util.*;

public class LeakyBucket {
    public static void main(String[] a){
        Scanner s=new Scanner(System.in);

        System.out.print("Bucket size: ");
        int b=s.nextInt();
        System.out.print("Output rate: ");
        int r=s.nextInt();
        System.out.print("Number of incoming packets: ");
        int n=s.nextInt();

        int stored = 0;

        for(int i=0;i<n;i++){
            System.out.print("Incoming packet size: ");
            int pkt=s.nextInt();

            if(pkt + stored <= b){
                stored += pkt;
                System.out.println("Bucket stored: " + stored);
            } else {
                System.out.println("Packet of size " + pkt + " DROPPED");
            }

            stored = Math.max(0, stored - r);
            System.out.println("After leaking: " + stored + "\n");
        }
    }
}

File: leakybucket.c
#include <stdio.h>  

int main() {
    int bucket_size, output_rate, num_packets;
    int stored = 0;

    printf("Enter bucket size: ");
    scanf("%d", &bucket_size);

    printf("Enter output rate: ");
    scanf("%d", &output_rate);

    printf("Enter the number of incoming packets: ");
    scanf("%d", &num_packets);

    for (int i = 0; i < num_packets; i++) {
        int packet_size;

        printf("Enter the incoming packet size: ");
        scanf("%d", &packet_size);

        if (packet_size + stored <= bucket_size) {
            stored += packet_size;
            printf("Bucket stored: %d\n", stored);
        } else {
            printf("Packet of size %d DROPPED\n", packet_size);
        }

        stored = (stored - output_rate > 0) ? stored - output_rate : 0;
        printf("After leaking: %d\n\n", stored);
    }

    return 0;
}